docs/book/src/developer/architecture/controllers/multi-tenancy.md

```diff
@@ -0,0 +1,12 @@
+# Multi tenancy
+
+Multi tenancy in Cluster API defines the capability of an infrastructure provider to manage different credentials, each
+one of them corresponding to an infrastructure tenant.
+
+## Contract
+In order to support multi tenancy, the following rule applies:
+
+- All the infrastructure provider are required to manage different credentials.
+- All the provider should deploy and run web-hook and with managers;
+  please note this is the default in controllers generated with kubebuilder. 
+- All the provider are required to create a `component.yaml` accordingly.
```

Comment:
My intent is to link documentation from https://github.com/kubernetes-sigs/cluster-api/pull/3985 here, depending on which PR is going to merge first